storagemarket/impl/providerstates/provider_states_test.go

```diff
@@ -305,7 +304,7 @@ func TestEnsureProviderFunds(t *testing.T) {
 			},
 			dealInspector: func(t *testing.T, deal storagemarket.MinerDeal) {
 				require.Equal(t, storagemarket.StorageDealProviderFunding, deal.State)
-				require.Equal(t, cids[0], deal.AddFundsCid)
+				require.Equal(t, &cids[0], deal.AddFundsCid)
```

Comment:
I am really confused about pointers, equality and how this works with serialization into and out of the store. I guess maybe in this case `require.Equal` is dereferencing the pointers under the hood?

I guess a more general question, why do we have a mix of `*cid.Cid` and `cid.Cid` in the `MinerDeal` and `ClientDeal`?  Should we be using one or the other?

Reply:
yes -- good point, non-obvious, but this is knowledge that should be shared by everyone. A CID that is not ALWAYS set should be a pointer. If a data structure has a regular cid with a value of cid.Undef, it will NOT serialize to CBOR. These changes are actually corrections to already merged code that had to be made once I ran our cbor generation scripts. I'm working. @shannonwells is working on a ticket to add checking that CBOR generation has been run to our backlog.